tests/chainermn_tests/optimizer_tests/test_double_buffering_optimizer.py

```diff
@@ -86,9 +86,15 @@ def test_update(self):
         chainer.testing.assert_allclose(
             self.optimizer.communicated_target.c.W.grad,
             (base + 5) * np.ones((5, 4)))
-        # barrier() requires before destructor of PureNcclCommunicator
-        # because communication may not be finished.
-        self.comm.mpi_comm.barrier()
+        self.comm.finalize()
```

Comment:
I think the current `comm.finalize` does not call `barrier`. Is it OK for this case?

Reply:
I think it is not OK. But, `barrier()` is needed in many cases.
I will add `barrier()` to `finalize()`